lib/ipfs/uploadToIpfs.ts

```diff
@@ -0,0 +1,49 @@
+import { hashFiles } from './hash'
+
+export type IPFSUploadResponse = {
+  cid: string
+  uri: string
+}
+
+const uploadCache = {
+  prefix: 'Pinata/IPFSUploadCache',
+  get(files: File[]): IPFSUploadResponse | undefined {
+    const digest = hashFiles(files)
+    try {
+      const cid = localStorage.getItem(`${this.prefix}/${digest}`)
+      if (cid) {
+        return { cid, uri: `ipfs://${cid}` }
+      }
+    } catch (error) {
+      console.error(error)
+    }
+  },
+  put(files: File[], cid: string) {
+    const digest = hashFiles(files)
+    try {
+      localStorage.setItem(`${this.prefix}/${digest}`, cid)
+    } catch (error) {
+      console.error(error)
+    }
+  },
+}
+
+export const uploadFile = async (file: File): Promise<IPFSUploadResponse> => {
+  try {
+    const data = new FormData()
+    data.set('file', file)
+    const cached = uploadCache.get([file])
+    if (cached) return cached
+    const res = await fetch('/api/ipfs', {
```

Comment:
You are referencing an API endpoint `/api/ipfs` but I do not see the files added.
- When do you plan to add the missing API endpoint???